prost_build/src/main.rs

```diff
@@ -0,0 +1,155 @@
+//! Protobuf files in the Sommelier repo, copying the result to the Sommelier_proto crate for import
+//! and use. While this builder generates about a dozen files only one contains all the Sommelier
+//! proto info and the rest are discarded in favor of upstream cosmos-sdk-proto
+
+// Building new Sommelier rust proto definitions
+// run 'cargo run'
+
+use regex::Regex;
+use std::{
+    ffi::OsStr,
+    fs::{self, create_dir_all, remove_dir_all},
+    path::PathBuf,
+};
+use std::{io, path::Path};
+use walkdir::WalkDir;
+
+/// Protos belonging to these Protobuf packages will be excluded
+/// (i.e. because they are sourced from `tendermint-proto` or `cosmos-sdk-proto`)
+const EXCLUDED_PROTO_PACKAGES: &[&str] = &["gogoproto", "google", "tendermint", "cosmos.base"];
+/// Regex for locating instances of `cosmos-sdk-proto` in prost/tonic build output
+const COSMOS_SDK_PROTO_REGEX: &str = "(super::)+cosmos";
+
+/// A temporary directory for proto building
+const TMP_PATH: &str = "/tmp/sommelier/";
+/// the output directory
+const OUT_PATH: &str = "../somm_proto/src/prost/";
+
+// All paths must end with a / and either be absolute or include a ./ to reference the current
+// working directory.
+
+fn main() {
+    let out_path = Path::new(&OUT_PATH);
+    let tmp_path = Path::new(&TMP_PATH);
+    compile_protos(out_path, tmp_path);
+}
+
+fn compile_protos(out_dir: &Path, tmp_dir: &Path) {
+    println!(
```

Comment:
nit: better for informational stuff flow out of stderr / eprintln to help people that pipe stuff (me)